### ISSUE TITLE
The metrics in wandb is abnormal

### ISSUE THREAD

**Issue by mst272** [huggingface/trl#1804]
I used the RLOOTrainer from trl/example for training. The results I saw on wandb after training are as follows. I really don't know what went wrong. Can someone help me? I don't understand why global_step is always 0.
![gitpic](https://github.com/huggingface/trl/assets/67250532/82888269-e73b-4aa5-97ac-4f9f819ffef3)
![gitpic2](https://github.com/huggingface/trl/assets/67250532/31e38028-11f2-464f-aed5-59d0acdb6bd7)

**Comment by mst272** [comment]
The following warnings will appear when running:

>/home/miniconda3/envs/code/lib/python3.9/site-packages/torch/autograd/graph.py:744: UserWarning: c10d::broadcast_: an autograd kernel was not registered to the Autograd key(s) but we are trying to backprop through it. This may lead to silently incorrect behavior. This behavior is deprecated and will be removed in a future version of PyTorch. If your operator is differentiable, please ensure you have registered an autograd kernel to the correct Autograd key (e.g. DispatchKey::Autograd, DispatchKey::CompositeImplicitAutograd). If your operator is not differentiable, or to squash this warning and use the previous behavior, please register torch::CppFunction::makeFallthrough() to DispatchKey::Autograd. (Triggered internally at ../torch/csrc/autograd/autograd_not_implemented_fallback.cpp:63.)
  return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass

**Comment by vwxyzjn** [comment]
I am not sure what happened to `global_step`. Could you try using `train/episode` or `Step` as the x-axis in wandb?

**Comment by mst272** [comment]
> I am not sure what happened to `global_step`. Could you try using `train/episode` or `Step` as the x-axis in wandb?
 i using Step as the x-axis in wandb，so my training was successful? Even if global_step is always zero.
![gitpic](https://github.com/huggingface/trl/assets/67250532/9415014b-b1dc-44c3-a1aa-60224c709e05)

**Comment by vwxyzjn** [comment]
Yeah... that's a bit weird

**Comment by mst272** [comment]
hi@vwxyzjn, thanks for the response! I'm really enjoying using the trl，this is a great work.
